modules/node-shared/src/test/scala/org/tessellation/node/shared/domain/seedlist/SnapshotOrdinalTimelineSuite.scala

```diff
@@ -0,0 +1,79 @@
+package org.tessellation.node.shared.domain.seedlist
+
+import cats.data.NonEmptyList
+import cats.syntax.all._
+
+import org.tessellation.node.shared.domain.seedlist.generators.terminatedOrdinalRangeGen
+import org.tessellation.node.shared.domain.seedlist.snapshotOrdinalTimeline.{SnapshotOrdinalTimeline, decode}
+
+import eu.timepit.refined.auto._
+import eu.timepit.refined.types.numeric.NonNegLong
+import org.scalacheck.Gen
+import weaver.SimpleIOSuite
+import weaver.scalacheck.Checkers
+
+object SnapshotOrdinalTimelineSuite extends SimpleIOSuite with Checkers {
+
+  test("parse input") {
+    val gen = for {
+      count <- Gen.chooseNum(1, 5)
+      b <- Gen.listOfN(count, terminatedOrdinalRangeGen)
+      c = b.tail.foldLeft(b.head._3) {
+        case (acc, (_, _, str)) =>
+          acc + ";" + str
+      }
+      ranges = b.map { case (a, b, _) => (a, b) }.map {
+        case (start, end) =>
+          Range(start, end)
+      }
+      entries = ranges
+        .sortBy(_.start.value)
+        .foldLeft(List.empty[Range]) {
+          case (prev :: tail, curr) if curr.start <= prev.end =>
+            val merged = Range(
+              prev.start,
+              NonNegLong.unsafeFrom(
+                Math.max(prev.end, curr.end)
+              )
+            )
+
+            merged :: tail
+          case (acc, curr) =>
+            curr :: acc
+        }
+        .reverse
+        .toNel
+        .get
+    } yield (entries, c)
+
+    forall(gen) {
+      case (entries, str) =>
+        val actual = decode(str).toEither
+        val expected = SnapshotOrdinalTimeline(entries).asRight[Throwable]
+
+        expect.same(expected, actual)
+    }
+  }
+
+  pureTest("parse empty input") {
+    val actual = decode("").toEither
+    val expected = NonEmptyList.of(InvalidOrdinal("")).asLeft[SnapshotOrdinalTimeline]
+
+    expect.same(expected, actual)
+  }
+
+  pureTest("merge overlapping ranges") {
+    val actual = decode("0-20;20-50;60-69;62-65;70-79").toEither
+
+    val expected = SnapshotOrdinalTimeline(
+      NonEmptyList.of(
+        Range(0L, 50L),
+        Range(60L, 69L),
+        Range(70L, 79L)
+      )
```

Comment:
I'm not seeing my previous comment on this issue; maybe it got lost.
Since `Range` is inclusive, couldn't we reduce this further to this?
```suggestion
    val expected = SnapshotOrdinalTimeline(
      NonEmptyList.of(
        Range(0L, 50L),
        Range(60L, 79L)
      )
```

Reply:
oh, actually I agree with @sdavidp.

range: `60-69;70-79` should be merged into `60-79` 🤔

Reply:
Per our offline discussion, this is a non-issue.

Reply:
Adjusted the merge case to

```scala
        case (prev :: tail, curr) if curr.start - prev.end <= 1 =>
```